syft/messaging/plan/plan.py

```diff
@@ -578,6 +580,100 @@ def detail(worker: AbstractWorker, plan_tuple: tuple) -> "Plan":
 
         return plan
 
+    @staticmethod
+    def bufferize(worker: AbstractWorker, plan: "Plan") -> PlanPB:
+        """
+        This function takes the attributes of a Plan and saves them in a Protobuf message
+        Args:
+            worker (AbstractWorker): the worker doing the serialization
+            plan (Plan): a Plan object
+        Returns:
+            PlanPB: a Protobuf message holding the unique attributes of the Plan object
+
+        """
+        protobuf_plan = PlanPB()
+
+        sy.serde.protobuf.proto.set_protobuf_id(protobuf_plan.id, plan.id)
+
+        protobuf_operations = [
+            sy.serde.protobuf.serde._bufferize(worker, operation).operation
+            for operation in plan.operations
+        ]
+        protobuf_plan.operations.extend(protobuf_operations)
+
+        protobuf_plan.state.CopyFrom(sy.serde.protobuf.serde._bufferize(worker, plan.state))
+
+        protobuf_plan.include_state = plan.include_state
+        protobuf_plan.is_built = plan.is_built
+        protobuf_plan.name = plan.name
+        protobuf_plan.tags.extend(plan.tags)
+
+        if protobuf_plan.description:
+            protobuf_plan.description = plan.description
+
+        if type(plan.placeholders) == type(dict()):
+            placeholders = plan.placeholders.values()
+        else:
+            placeholders = plan.placeholders
+
+        protobuf_placeholders = [
+            sy.serde.protobuf.serde._bufferize(worker, placeholder) for placeholder in placeholders
+        ]
+        protobuf_plan.placeholders.extend(protobuf_placeholders)
+
+        return protobuf_plan
+
+    @staticmethod
+    def unbufferize(worker: AbstractWorker, protobuf_plan: PlanPB) -> "Plan":
+        """This function reconstructs a Plan object given its attributes in the form of a Protobuf message
+        Args:
+            worker: the worker doing the deserialization
+            protobuf_plan: a Protobuf message holding the attributes of the Plan
+        Returns:
+            plan: a Plan object
+        """
+
+        worker._tmp_placeholders = {}
+        id = sy.serde.protobuf.proto.get_protobuf_id(protobuf_plan.id)
+
+        operations = []
+        for operation in protobuf_plan.operations:
+            op_msg = OperationMessagePB()
```

Comment:
Why are we creating Protobuf objects in `unbufferize`? The Protobuf schemas maintain a distinction between operations (without regard to where they might be executed) and operation messages (which contain operations being sent somewhere.) That distinction doesn't exist in Python yet, which means that `_unbufferize` doesn't know how to handle raw `OperationPB` objects yet. I think this should be cleaned up, but maybe as part of further refactoring the `Operation` class.